apps/site/src/app/admin/participants/components/ParticipantAction.tsx

```diff
@@ -1,26 +1,68 @@
+import { useContext } from "react";
+
 import Button from "@cloudscape-design/components/button";
 
+import UserContext from "@/lib/admin/UserContext";
+import { isCheckinLead } from "@/lib/admin/authorization";
+import { Decision, PostAcceptedStatus } from "@/lib/admin/useApplicant";
 import { Participant } from "@/lib/admin/useParticipants";
+import ParticipantActionPopover from "./ParticipantActionPopover";
 
 interface ParticipantActionProps {
 	participant: Participant;
 	initiateCheckIn: (participant: Participant) => void;
+	initiatePromotion: (participant: Participant) => void;
 }
+
 function ParticipantAction({
 	participant,
 	initiateCheckIn,
+	initiatePromotion,
 }: ParticipantActionProps) {
-	// TODO: waitlist promotion
+	const { role } = useContext(UserContext);
+
+	const isCheckin = isCheckinLead(role);
+	const isWaiverSigned = participant.status === PostAcceptedStatus.signed;
+
+	const promoteButton = (
+		<Button
+			variant="inline-link"
+			ariaLabel={`Promote ${participant._id} off waitlist`}
+			onClick={() => initiatePromotion(participant)}
+			disabled={isCheckin}
```

Comment:
Issue [repeating]: since we negated the source variable, we'll have to negate the condition used here and on line 50.